src/Modules/Manage/Dnn.PersonaBar.Users.Tests/UserValidatorUnitTests.cs

```diff
@@ -0,0 +1,159 @@
+        {
+            // Arrange
+            SetupForSiteGroup(true);
+            InitializeUserValidator();
+
+            // Act
+            var result = _userValidator.ValidateUser(_userId, _portalSettings, null, out _userInfo);
+
+            // Assert
+            Assert.IsNull(result);
+        }
+
+
+        [Test]
+        public void ValidateUser_IfUserNotAllowedInSiteGroup_ThenErrorResponse()
+        {
+            // Arrange
+            SetupForSiteGroup(false);
+            InitializeUserValidator();
+
+            // Act
+            var result = _userValidator.ValidateUser(_userId, _portalSettings, null, out _userInfo);
+
+            // Assert
+            Assert.IsTrue(result.IsError);
+        }
+
+        private void SetupForSiteGroup(bool isAllowed)
+        {
+            _userInfo = null;
+            SetupUserControllerWrapperMock();
+
+            var otherPortalId = 2;
+            var portals = new ArrayList();
+            portals.Add(new PortalInfo() { PortalID = otherPortalId });
+            _portalControllerMock.Setup(p => p.GetPortals()).Returns(portals);
+
+            SetUserInfoWithProfile();
+            _userControllerWrapperMock.Setup(u => u.GetUserById(It.IsAny<int>(), It.IsAny<int>())).Returns(_userInfo);
+            _contentVerifierMock.Setup(c => c.IsContentExistsForRequestedPortal(It.IsAny<int>(), It.IsAny<PortalSettings>(), true)).Returns(isAllowed);
+        }
+
+        [Test]
+        public void ValidateUser_IfUserIdNotFound_ThenErrorResponse()
+        {
+            // Arrange
+            _userInfo = null;
+            SetupUserControllerWrapperMock();
+
+            ArrayList portals = new ArrayList();
+            _portalControllerMock.Setup(p => p.GetPortals()).Returns(portals);
+
+            InitializeUserValidator();
+
+            // Act
+            var result = _userValidator.ValidateUser(_userId, _portalSettings, null, out _userInfo);
+
+            // Assert
+            Assert.IsTrue(result.IsError);
+        }
+
+        [Test]
+        public void ValidateUser_IfUserIdWithoutValue_ThenErrorResponse()
+        {
+            // Arrange
+            InitializeUserValidator();
+
+            _userId = null;
+
+            // Act
+            var result = _userValidator.ValidateUser(_userId, _portalSettings, null, out _userInfo);
+
+            // Assert
+            Assert.IsTrue(result.IsError);
+        }
+
+        private void InitializeUserValidator()
+        {
+            _userValidator = new UserValidator(
+                _portalControllerMock.Object,
+                _userControllerWrapperMock.Object,
+                _contentVerifierMock.Object
+            );
+        }
+
+        private void SetUserInfoWithProfile()
+        {
+            _userInfo = new UserInfo();
+            var profile = new UserProfile();
+            profile.FirstName = "testUser";
+            _userInfo.UserID = _userId.Value;
+            _userInfo.Profile = profile;
+        }
```

Comment:
I think it will be cleaner if we avoided methods with side effects. Please change this one to be `UserInfo SetUserInfoWithProfile()`